src/ControlSurface/ScoreControlSurfaceList.qml

```diff
@@ -1,41 +1,79 @@
 import QtQuick 2.0
 import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.2
 import QtQuick.Controls.Styles 1.4
 import QtQml.Models 2.12
 
-ListView {
+Item  {
     id: scoreControlSurfaceList
-    spacing: 20
     anchors.left: parent.left
     anchors.margins: 5
     anchors.right: parent.right
     anchors.top: parent.top
     anchors.bottom: parent.bottom
     height: parent.height
-    orientation: ListView.Vertical
-    clip: true
-    snapMode: ListView.SnapToItem
 
-    // ObjectModel's "model" data is the actual item that is going to be displayed
-    model: ObjectModel {
-        id: controlSurfacelist
-    }
+    ListView {
+        id: scoreControlSurfaceListView
+        spacing: 20
+        anchors.left: parent.left
+        anchors.margins: 5
+        anchors.right: scrollBar.left
+        height: parent.height
+        orientation: ListView.Vertical
+        clip: true
+        snapMode: ListView.SnapToItem
+
+        // ObjectModel's "model" data is the actual item that is going to be displayed
+        model: ObjectModel {
+            id: controlSurfacelist
+        }
 
-    // The Component will serve as a factory to create the individual ControlSurface visual items & data model
-    Component {
-        id: controlSurfaceDataModel
-        Loader {
-            id: controlSurface
-            property string name
-            property var path
-            property var m
-            signal appendControls(var m)
-            signal modifyControl(var m)
-            source: "ScoreControlSurface.qml"
-            onLoaded: {
-                controlSurface.appendControls(m)
+        // The Component will serve as a factory to create the individual ControlSurface visual items & data model
+        Component {
+            id: controlSurfaceDataModel
+            Loader {
+                id: controlSurface
+                width: scoreControlSurfaceListView.width
+                property string name
+                property var path
+                property var m
+                signal appendControls(var m)
+                signal modifyControl(var m)
+                source: "ScoreControlSurface.qml"
+                onLoaded: {
+                    controlSurface.appendControls(m)
+                }
             }
         }
+
+        ScrollBar.vertical: scrollBar
+    }
+
+    ScrollBar {
+            id: scrollBar
+            active: true
+            visible: scoreControlSurfaceListView.count > 0
+            width: window.width <= 500 ? 20 : 30
+            height: parent.height
+
+            anchors.right: parent.right
+            policy: ScrollBar.AsNeeded
+            snapMode: ScrollBar.SnapAlways
+            contentItem: Rectangle {
+                id: scrollBarContentItem
+                visible: scrollBar.size >= 1 ? false : true
```

Comment:
equivalent to 

    visible: scrollBar.size < 1

(but it's weird, no ?)

Reply:
changed !